### DISCUSSION TITLE
m.AddEndpoint micro service how to add pub sub pattern

### DISCUSSION THREAD

**Discussion by khanakia** [nats-io/nats.go#1596] [Q&A]
```go
m.AddEndpoint(
  "ping",
  micro.HandlerFunc(func(req micro.Request) {
	  req.Respond(req.Data())
  }),
  micro.WithEndpointMetadata(map[string]string{
	  "format": "application/json",
  }),
)
```

How do I add a pub sub pattern with micro service I do not want my client to wait for the reply but I do not see microservice has any endpoint to build something like that below
```go
// nats pub hello world
ec.Subscribe("hello", func(s string) {
	fmt.Printf("Received a message: %s\n", s)
})

// nats request help ""
ec.Subscribe("help", func(subj, reply string, msg string) {
	ec.Publish(reply, "I can help!")
})
```

**Comment by aricart** [comment]
Just to clarify - are asking for a way of getting a request message but to not respond to it?

**Reply by khanakia** [reply]
Yes, exactly so the client won't wait for it. just exactly Pub-Sub pattern `nats pub hello world`
